### ISSUE TITLE
SnD GUI miniwindow has truncated text on high DPI displays

### ISSUE THREAD

**Issue by danj2k** [AardCrowley/Search-and-Destroy#108]
On high DPI displays, the title bar and button text for the SnD GUI miniwindow appears truncated:

<img width="300" alt="image" src="https://github.com/AardCrowley/Search-and-Destroy/assets/29895048/8c49960f-e761-41cd-a712-c61f2cb68954">

If it helps, it looks like MUSHclient has a function GetDeviceCaps which allows checking the display DPI.

**Comment by danj2k** [comment]
You could fix this by multiplying hardcoded width and height values by (horizontal dpi / 96) and (vertical dpi / 96) respectively.

You can get the horizontal DPI with `GetDeviceCaps(88)` and vertical DPI with `GetDeviceCaps(90)` - see https://www.mushclient.com/mushclient/mw_other.htm#GetDeviceCaps

The changes would be mostly in `xg_draw_window()` and related drawing functions.

**Comment by AardCrowley** [comment]
Yeah, I had mentioned that on ftalk a half hour after your message. Said I've got that set up in v6 already. Hadn't forgotten it, just didn't want to go through the current code and make every change needed at the time. But that's exactly what I have.

Sent from Proton Mail Android

-------- Original Message --------

**Comment by AardCrowley** [comment]
Implemented in commit b321b79, using exactly the GetDeviceCaps(88) approach @danj2k suggested. The plugin already had a resolution-based auto-scale heuristic (_scale = _screen_w / 1920) driving font size selection, but that under-detects DPI scaling on a normal-resolution display (a 1920x1080 panel at 150% still reports 1920 wide) — which is exactly the clipping case in the original report. compute_scale() now takes whichever of resolution-ratio or DPI-ratio calls for more scale, so it also self-corrects for large 4K-and-up displays at 100% DPI without double-scaling. Not applied if you've manually picked a font/size via "Change Font..." — that's respected as-is.
